### PR TITLE
 Add projectsetup for react on rails as a one project with webpack

### PR DESCRIPTION
**Pull request by codeWithHyder**
- Create a new Rails app called 'hello-rails-react'.
- Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
- Verify that the jsbundling-rails gem is added to your Gemfile.
- Initialize your project with Git.
- Install needed node dependencies as described in the previous lesson's tutorial.
- Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- Make sure that your project has a Postgres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- Create a static view that will be the root of your app.

### CODE REVIEW

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# Required Changes 🔁
## Hello @codeWithHyder   👋 🙂

You're doing a great job on your project. We're proud of you. Good job so far! 

It's OK to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job. 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
  ✅  Correct use of Flow
  ✅  Descriptive Pull Request
  ✅  No linter errors

## Required Changes ♻️

- [x] As part of the project requirement, there should be a redux store, an action, and a reducer that uses the URL to the endpoint from the Rails API.



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -11,7 +11,25 @@
...
+      import store from '../store/store';
+      // Provide the store to your React app
+      import { Provider } from 'react-redux';
+      import React from 'react';
+      import ReactDOM from 'react-dom';
+      import App from '../components/App';
+
+      document.addEventListener('DOMContentLoaded', () => {
+        ReactDOM.render(
+          <Provider store={store}>
```
Comment (brytebee) [inline app/views/layouts/application.html.erb:27]: 👏👏👏👏👏

**Review by brytebee — APPROVED** [review]
Hi, @codeWithHyder 👋,

##

### Highlights💡
- Great adding redux store as suggested by the previous reviewer.✔️
- Good commit message.✔️ 
- Good PR Summary.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
- N/A

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (codeWithHyder) [reply app/views/layouts/application.html.erb:27]: Thank you CR

**Pull request merged**